### PR TITLE
Functionality - auth (login , sign up, api implemented, validations implemented)

### PR DESCRIPTION
**Pull request by kgajriaAgarwal**
Functionality - auth (login , sign up, api implemented, validations implemented)
Preview: https://blithe-la-nota.netlify.app/
files to check validations folder, Auth folder, services folder

### CODE REVIEW

File: src/Helpers/Services/actions.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    return response;
+  };
+  
+  // Action to perform Signup
+  export const actionSignup = async (data) => {
+    const response = await mainApiService("signup", data);
+    if (response.data.encodedToken) {
+      setLocalStorage("authData", response.data.encodedToken);
+    }
+    // dispatch({ type: Actions.LOGIN, data: response.data });
```
Comment (MayankKumar10) [inline src/Helpers/Services/actions.js]: Suggestion: remove this comment

File: src/Helpers/Services/apiService.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+//   let userInfo = await getLocalStorage('authData');
+
+//   if (userInfo) {
+//     if (userInfo.access_token) {
+//       _headers_['Authorization'] = `bearer ${userInfo.access_token}`
+//     }
+//   }
+
+//   return _headers_;
+// };
```
Comment (MayankKumar10) [inline src/Helpers/Services/apiService.js:70]: Suggestion: remove this comment

File: src/Helpers/Services/apiService.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  }  
+ 
+  let requestObject = Object.assign({}, apiDetails);
+  requestObject.data = prepareDataObject(requestObject.data, data);
+  requestObject.url = injectParamsToUrl(requestObject.url, data);
+  requestObject.headers = await injectHeadersToReq(requestObject.headers, data);
+
+  return axios(requestObject)
+    .then(function (result) {
+        console.log("resulttt@@:", result)
```
Comment (MayankKumar10) [inline src/Helpers/Services/apiService.js:115]: Suggestion: remove this console.log

File: src/Pages/NotesPage/NotesPage.css [hunk 4]
```diff
@@ -1,4 +1,12 @@
 
+.main-container-wrapper{
+    /* background-color: cadetblue; */
+    width: '70%',;
```
Comment (MayankKumar10) [inline src/Pages/NotesPage/NotesPage.css]: Suggestion: remove this apostrophe from width.

**Review by MayankKumar10 — APPROVED** [review]
Look into these Suggestions everything else looks great you can merge it.

**Review by mikesch803 — APPROVED** [review]
just remove the commented lines, everything is good

**Review by Naman-Saxena1 — APPROVED** [review]
When we login with test credentials, menu option should change to "Logout"
Otherwise, code looks good 👍 

Capture1

File: src/Helpers/Services/apiService.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+  }
+}; 
+
+const mainApiService = async (apiKeyName, data) => {
+//   let apiDetails = ApiJson[apiKeyName];
+    let apiDetails = ApiJson[apiKeyName]
+  if (!apiDetails) {
+    console.log(
+      'Api configuration do not found in api-json, please check api-json.js'
+    );
```
Comment (sri-rishi) [inline src/Helpers/Services/apiService.js:101]: Remainder: remove console's from code after done working

File: src/Helpers/Validations/ValidationRegex.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+export const validateRegex ={
+    alphaOnly: '^[a-zA-Z .]*$', // alphabet space and dot
+    //alphaOnly: '^[a-zA-Z ]*$',// alphabet and space
+    //alphaOnly: '/^[A-Za-z\\s]+$/',
+    // passwordPattern:
+    //   '/^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[$@$!%*?&#])[A-Za-z\\d$@$!%*?&#]{8,20}/',
+  
+    // passwordPattern:
+    //   '/^(?=\\S*[a-z])(?=\\S*[A-Z])(?=\\S*\\d)(?=\\S*[^\\w\\s])\\S{8,}$/',
```
Comment (sri-rishi) [inline src/Helpers/Validations/ValidationRegex.js:9]: Suggestions: remove commented code from files

**Review by sri-rishi — APPROVED** [review]
Nice work , code looks good to merge

File: src/Helpers/Services/apiService.js (hunk 2 above)
Comment (kgajriaAgarwal) [reply src/Helpers/Services/apiService.js:70]: removed

File: src/Helpers/Services/apiService.js (hunk 3 above)
Comment (kgajriaAgarwal) [reply src/Helpers/Services/apiService.js:115]: //code is commented in this file for future perspective..
////console.log is also commented , until the time the this app is fully developed

File: src/Pages/NotesPage/NotesPage.css (hunk 4 above)
Comment (kgajriaAgarwal) [reply src/Pages/NotesPage/NotesPage.css]: removed

File: src/Helpers/Services/apiService.js (hunk 5 above)
Comment (kgajriaAgarwal) [reply src/Helpers/Services/apiService.js:101]: ok

File: src/Helpers/Validations/ValidationRegex.js (hunk 6 above)
Comment (kgajriaAgarwal) [reply src/Helpers/Validations/ValidationRegex.js:9]: will remove it after complete development

**Pull request merged**